tests/test_giturl.py

```diff
@@ -0,0 +1,74 @@
+# fetchcode is a free software tool from nexB Inc. and others.
+# Visit https://github.com/nexB/fetchcode for support and download.
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# http://nexb.com and http://aboutcode.org
+#
+# This software is licensed under the Apache License version 2.0.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: 
+# http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import os
+from unittest import mock
+
+from fetchcode import giturl
+
+@mock.patch('os.system')
+def test_https(mock_os):
+    """
+    Testing https based URLs
+    """
+    url = 'https://github.com/TG1999/fetchcode.git'
+    with mock.patch('os.mkdir') as mocked_file:
+        response = giturl.fetch(source=url,branch='master',depth=None,dest='/home/tg')
+        assert response is not None
+
+
+@mock.patch('os.system')
+def test_git(mock_os):
+    """
+    Testing git based URLs
+    """
+    url = 'git://github.com/jamesor/mongoose-versioner'
+    with mock.patch('os.mkdir') as mocked_file:
+        response = giturl.fetch(source=url,branch='master',depth=None,dest='/home/tg')
+        assert response is not None
+
+
+@mock.patch('os.system')
+def test_git_ssh(mock_os):
+    """
+    Testing git+ssh based URLs
+    """
+    url = 'git+ssh://github.com/bcoe/thumbd.git'
+    with mock.patch('os.mkdir') as mocked_file:
+        response = giturl.fetch(source=url,branch='master',depth=None,dest='/home/tg')
+        assert response is not None
+
+
+@mock.patch('os.system')
+def test_git_https(mock_os):
+    """
+    Testing git+https based URLs
+    """
+    url = 'git+https://github.com/Empeeric/i18n-node'
+    with mock.patch('os.mkdir') as mocked_file:
+        response = giturl.fetch(source=url,branch='master',depth=None,dest='/home/tg')
+        assert response is not None
+
+
+@mock.patch('os.system')
+def test_git_https(mock_os):
+    """
+    Testing ssh based URLs
```

Comment:
test_git_https makes me believe that this is a test for git and https
But the docstring says "Testing ssh based URLs"
It would be simpler and cleaner to:
1. use verbose and descriptive test function names
2. do not use docstring for tests

So your function could be instead:
`test_fetch_with_ssh_git_url_returns_a_response(mock_os)`